[Step6 ~ Step8] Terry-yes

terry-yes opened the pull request:
`8. 내비게이션-컨트롤러`까지 구현했습니다.

- 프로필 수정화면에서 기기내의 이미지를 고를 수 있도록 구현(ImagePickerController사용)
- 로그인 화면을 구현하였습니다.
- 로그인을 성공하면 웹뷰화면으로 연결됩니다. 
  - WKWebView를 사용하여 구현하였습니다.

- 전체적인 구현 화면은 README.md파일에서 확인하실 수 있습니다.([링크](https://github.com/terry-yes/w2-signup-profile/tree/step6))

terry-yes committed:
chore: UserInterfaceState.xcuserstate파일 gitignore추가

terry-yes committed:
feat: 프로필 수정화면의 이미지를 ImagePickerController로 선택할수 있게 변경

terry-yes committed:
feat: 프로필 수정화면에서 변경된 이미지가 프로필화면에 적용되도록 구현

terry-yes committed:
feat: 프로필 화면에 오토 레이아웃 적용

terry-yes committed:
feat: 로그인 화면 구성 및 오토 레이아웃 적용

terry-yes committed:
feat: login화면에서 id와 pw가 "kakao"일때만 다음 화면으로 넘어가도록 구현

terry-yes committed:
feat: MainVC에서 웹뷰가 작동되도록 구현

- 주소는 https://page.kakao.com

terry-yes committed:
docs: 로그인 화면, 웹뷰 구현까지 README.md파일에 정리

timothy-podo commented on KakaoProfile/KakaoProfile/EditViewController.swift:
```diff
@@ -32,9 +33,12 @@ class EditViewController: UIViewController {
-        let name = nameTextField.text ?? ""
-        let description = descriptionTextField.text ?? ""
-        profileDataDelegate?.updateProfile(name: name, description: description)
+        let name = self.nameTextField.text ?? ""
```
1. 여기에 self 를 명시적으로 추가하신 이유가 있을까요?
2. default value 로 empty string 을 사용하신 이유가 있을까요?

terry-yes replied:
티모시 안녕하세요! 리뷰해주셔서 감사합니다!!😄

1. 사실 원래 self를 표기하지 않았었는데요. 메소드에서 변수를 사용할 때 타입 프로퍼티에서 가져온 변수인지 메소드내에서 생성된 변수인지 알면 코드를 읽기 편하지 않을까라는 생각이 들어서 한번 명시해봤습니다. 
쓰다보니 단점이 발견되었는데요. nameTextField, *Label 등 view에 관련된 타입 프로퍼티는 이미 변수명에서 타입 프로퍼티인게 구분이 되더라고요. 그래서 굳이 self를 쓰는게 맞을까 하는 생각이 들어서 사실 없앨까 고민하고 있었습니다.
지금은 일괄적으로 표기하는쪽으로 하였는데요. 혹시 티모시는 어떻게 생각하시는지 말씀해주시면 반영하겠습니다!

2. 여기는 제가 잘못 생각한 것 같습니다.😅  텍스트필드에 아무런 값이 없는 경우가 nil이라고 생각했엇는데, 확인해보니 값이 없으면 빈문자열이 리턴 되더라고요. 말씀해주신 부분과 비슷한 부분 모두 옵셔널 바인딩으로 변경하였습니다!

timothy-podo replied:
1. self 표기와 관련한 정답이 있어서 말씀드린것은 아니고, 테리 의견이 궁금하여서 여쭤봤어요 :) 페이지에서는 누가봐도 클래스의 프로퍼티라고 판단되는건 암묵적으로 self 표기를 안하는 방향으로  작업을 하고 있긴 하지만, 다른 팀은 다른 룰을 가지고 계실 수도 있어서 뭐가 맞다고 할 수는 없을거 같아요. 
2. 옵셔널 바인딩 말씀을 드리려던 것은 아니었고 ㅎㅎㅎ 지금 보면 프로토콜에서 전달받는 파라미터가 required 로 되어있으니 당연히 default valueu 로 empty string 을 쓰셨다고 생각 되었는데요. 프로토콜의 구현체 내부를 보니 UILabel 의 text 또는 UIImage 를 셋팅하고 계셔서, 그 값들은 이미 optional 을 받아주고 있기 때문에 프로토콜의 파라미터가 optional 로 선언되어있다면 이쪽에서 optional binding 혹은 default value 처리 해 줄 필요가 없지 않을까 싶은 생각이 들어요.

terry-yes replied:
1. 아하 그렇군요!! 이번 프로젝트는 이렇게 한번 해보고 다음 프로젝트에서는 티모시가 말씀해주신 방법으로 해보겠습니다!! 둘 다의 장단점을 느껴보겠습니다. ㅎㅎ
2. 감사합니다. 티모시가 말씀해주신 방법은 생각해보지 못했던것 같아요. 옵셔널에서 옵셔널로 넘어가는데 굳이 중간에 옵셔널을 unwrapping할 필요가 없었네요! 뭔가 옵셔널은 항상 unwrapping하려는 생각이 있었던 것 같습니다.
말씀해주신 대로 프로토콜과 구현체에서 옵셔널을 전달받을 수 있도록 변경하였습니다!

timothy-podo commented on KakaoProfile/KakaoProfile/EditViewController.swift:
```diff
@@ -32,9 +33,12 @@ class EditViewController: UIViewController {
+        let description = self.descriptionTextField.text ?? ""
+        guard let image = self.profileImage.image else { return }
+        //선택된 이미지가 없을때 done버튼이 수행이 안되는게 맞을까?
+          // 사진이 없다고 알림을 주면서 dismiss하는게 자연스러울까?
```
인덴테이션이 맞으면 좋겠네요 :)

terry-yes replied:
수정하였습니다!

timothy-podo commented on KakaoProfile/KakaoProfile/LoginViewController.swift:
```diff
@@ -0,0 +1,29 @@
+            passwordTextField.text == "kakao" {
+            return true
+        }
+        return false
```
삼항연산자를 쓰는건 어떨까요?

terry-yes replied:
수정하였습니다!😀
리뷰 감사드립니다! 좋은 하루 되세요~

```swift
override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
    return self.idTextField.text == "kakao" && self.passwordTextField.text == "kakao" ? true : false
}
```

terry-yes committed:
refactor: 리뷰 사항 반영

- nameTextFiled.text등이 nil일 때 기본값을 주던것을 옵셔널 바인딩으로 변경
- view에 관련된 타입 프로퍼티를 호출하는 경우 self를 전부 명시
- 삼항연산자 반영
- 자잘한 인덴트 맞춤, 빈줄 제거

terry-yes committed:
refactor: 코드 리뷰 사항 반영

- 프로필 수정화면에서의 뷰들을 델리게이트의 구현체로 보낼때 'unwrapping하지 않은 옵셔널 상태'로 전달되게 변경

timothy-podo commented on KakaoProfile/KakaoProfile/EditViewController.swift:
```diff
@@ -23,18 +24,24 @@ class EditViewController: UIViewController {
     func updateUI() {
-        nameTextField.text = nameText ?? ""
-        descriptionTextField.text = descriptionText ?? ""
+        guard let name = nameText,
+               let description = descriptionText
+        else {
+            return
+        }
+        self.nameTextField.text = name
+        self.descriptionTextField.text = description
```
위에 드린 코멘트와 동일하게, 아마 UITextField 의 text 도 optional value 를 받아주고 있을거에요.
https://developer.apple.com/documentation/uikit/uitextfield/1619635-text

guard 할필요 없이 바로 nameText, descriptionText 를 각각 assign 해 주셔도 되지 않을까 싶습니다.

terry-yes replied:
아하 그렇군요! ㅎㅎ
감사합니다. 수정하였습니다~

terry-yes committed:
refactor: 코드 리뷰  사항 반영

- textFIeld.text에 변수를 넘겨줄 때 옵셔널 바인딩하던 것을 하지 않는쪽으로 변경

timothy-podo reviewed (approved):
작업하신 사항 확인했습니다~ 
저는 별도로 코멘트 드릴거 없을거 같네요 :)

bamiboo-han commented on KakaoProfile/KakaoProfile/LoginViewController.swift:
```diff
@@ -0,0 +1,21 @@
+    }
+    
+    override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
+        return self.idTextField.text == "kakao" && self.passwordTextField.text == "kakao" ? true : false
```
이 부분은 return self.idTextField.text == "kakao" && self.passwordTextField.text == "kakao"
로 더 간단히도 표현할 수 있습니다!

terry-yes replied:
감사합니다!! 해당 사항 수정했습니다.😄

terry-yes committed:
refactor: 코드 리뷰사항 반영

- 삼항연산자 부분에서 true, false만 리턴하는 경우 true, false부분을 삭제함
